### ISSUE TITLE
Drop Python 3.3 support

### ISSUE THREAD

**Issue by underyx** [coala-analyzer/coala#2288]
Python 3.3 never had a significant user base and wasn't part of any distribution's LTS release.

**Comment by Makman2** [comment]
Are there any new modules we can use functionality from? if not there's no need to imo^^
(there's only one case I remember, and this is `partial_method` which is copied into our coala future module^^)

**Comment by underyx** [comment]
@Makman2 Well I mean, dropping support doesn't mean it has to stop working, it's just that we will stop checking if it works.

**Comment by Makman2** [comment]
Feel free to make a cEP :P

**Comment by sils1297** [comment]
@underyx so currently we don't feel any big pain of supporting 3.3, 3.2 was far worse. Any particular reason?

**Comment by underyx** [comment]
@sils1297 @Makman2 are you actually testing if the code still works on 3.3?

**Comment by sils1297** [comment]
CI does that

**Comment by underyx** [comment]
So we could drop 33% of the builds and decrease CI queue time by making a change that won't affect basicaly anyone 🎉

**Comment by AbdealiJK** [comment]
FWIW: If we drop 3.3 support, our circleci build queue time decreases by 50% as we will only use 2 of the 4 nodes available.

TLDR:
If there are 2 pushes consecutively, both the builds (each having 2 jobs) can be executed simultaneously because we have a quota of 4 nodes for circleci. In our current case where we use 3, the 4th node from circleci doesn't get used at all, because circleci doesn't start a build unless all the nodes required for the build are available.
So, dropping 3.3 would give us a 50% decrease in queue time in CircleCI (and appveyor anyway doesnt test 3.3)

PS: I don't really have an opinion on whether 3.3 should be dropped. Python 3.2 and 3.3 are technically still "supported" by python.org

**Comment by underyx** [comment]
@AbdealiJK you wrote a one line comment and then followed it up with a five line TLDR version :D

The build time decrease sounds awesome to me. And [there's a bunch of libraries dropping 3.3 support](https://google.com/search?q="Python 3.3 never had a significant user base and wasn't part of any distribution's LTS release") regardless of Python's stance.

**Comment by Makman2** [comment]
Python 3.3 is officially dropped. The above improvements should be still implemented :+1:

**Comment by sils1297** [comment]
They are already implemented in the dropping commits.

**Comment by Makman2** [comment]
Indeed, have missed that :+1:
